src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,34 +36,111 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        this.blockName = new Block(extractBlockNameFromAddressString(trimmedAddress));
+        this.streetName = new Street(extractStreetNameFromAddressString(trimmedAddress));
+        this.unitName = new Unit(extractUnitNameFromAddressString(trimmedAddress));
+        this.postalCode = new PostalCode(extractPostalCodeFromAddressString(trimmedAddress));
+    }
+
+    /**
+     * Retrieves the block name based on the trimmedAddress
+     *
+     * @param trimmedAddress trimmed and valid address
+     * @return blockName of the address
+     */
+    private String extractBlockNameFromAddressString(String trimmedAddress){
+        String[] splitString = trimmedAddress.split(", ");
+        return splitString[ADDRESS_BLOCK_INDEX];
+    }
+
+    /**
+     * Retrieves the street name based on the trimmedAddress
+     *
+     * @param trimmedAddress trimmed and valid address
+     * @return
+     */
+    private String extractStreetNameFromAddressString(String trimmedAddress){
+        String[] splitString = trimmedAddress.split(", ");
+        return splitString[ADDRESS_STREET_INDEX];
+    }
+
+    /**
+     * Retrieves the unit name based on the trimmedAdress
+     *
+     * @param trimmedAddress trimmed and valid address
+     * @return
+     */
+    private String extractUnitNameFromAddressString(String trimmedAddress){
+        String[] splitString = trimmedAddress.split(", ");
+        return splitString[ADDRESS_UNIT_INDEX];
     }
 
+    /**
+     * Retrieves the postal code based on the trimmedAddress
+     *
+     * @param trimmedAddress trimmed and valid address
+     * @return
+     */
+    private String extractPostalCodeFromAddressString(String trimmedAddress){
+        String[] splitString = trimmedAddress.split(", ");
+        return splitString[ADDRESS_POSTAL_CODE_INDEX];
+    }
+
+
+
     /**
      * Returns true if a given string is a valid person address.
      */
-    public static boolean isValidAddress(String test) {
+    private static boolean isValidAddress(String test) {
```

Comment:
Why private here? What if another class wants to check if a given string is a valid address?